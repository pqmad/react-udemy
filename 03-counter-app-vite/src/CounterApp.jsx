import {useState} from 'react'
import PropTypes from "prop-types";


export const CounterApp = ({value}) => {
  const [counter, setcounter] = useState (value);


  const handleAdd = ()=>{
    setcounter(counter+1);
  }
  const handleSubs = ()=>{
    setcounter(counter-1);
  }
  const handleReset = ()=>{
    setcounter(value);
  }
  return (
    <>
        <h1>CounterApp</h1>
        <h2> { counter } </h2>
        <button aria-label='sumar' onClick={handleAdd}>+1</button>
        <button aria-label='restar' onClick={handleSubs}>-1</button>
        <button aria-label='reinicar' onClick={handleReset}>reset</button>
    </>
  )
}

CounterApp.propTypes = {
    value: PropTypes.number.isRequired
}
  
