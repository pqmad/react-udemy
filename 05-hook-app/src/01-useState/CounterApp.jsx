import { useState } from "react"


export const CounterApp = () => {
    // const [counter, setcounter] = useState({
    //     c1: 0,
    //     c2: 5,
    //     c3: 10
    // })

    const [counter, setcounter] = useState({
        c1: 0,
        c2: 5,
        c3: 10
    })
    const {c1,c2,c3}=counter;
  return (
    <>
        <h1>Counter1: {c1}</h1>
        <h1>Counter2: {c2}</h1>
        <h1>Counter3: {c3}</h1>
        <hr/>
        {/* <button className="btn" onClick={()=>setcounter({c1:c1+1,c2:c2,c3:c3})}>+1</button> */}
        <button className="btn" onClick={()=>setcounter({...counter,c1:c1+1})}>+1</button>
    </>
  )
}
