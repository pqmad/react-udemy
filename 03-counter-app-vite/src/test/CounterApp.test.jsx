import { fireEvent, render,screen } from "@testing-library/react"
import { CounterApp } from "../CounterApp"

describe('CounterApp tarea', () => { 
    
    test('debe hacer match con el snapshot', () => { 
        render( <CounterApp value={100}/> )
        expect(screen.container).toMatchSnapshot();
     });

     test('debe mostrar valor inicial de 100', () => { 
        render( <CounterApp value={100}/> )
        expect(screen.getByText(100)).toBeTruthy();
     });

 })

 describe('CounterApp', () => { 
    const valorinicial=10;
    
    test('debe incremetar con el botón +1', () => { 
        render( <CounterApp value={valorinicial}/> )
        fireEvent.click(screen.getByText("+1"))
        expect(screen.getByText("11")).toBeTruthy();
     });

     test('debe disminuir con el botón -1', () => { 
        render( <CounterApp value={valorinicial}/> )
        fireEvent.click(screen.getByText("-1"))
        expect(screen.getByText("9")).toBeTruthy();
     });
     test('debe funcionar el botón Reset', () => { 
      render( <CounterApp value={valorinicial}/> )
      fireEvent.click(screen.getByRole('button', {name:'reinicar'}));
      expect(screen.getByText(valorinicial)).toBeTruthy();
   });

 })