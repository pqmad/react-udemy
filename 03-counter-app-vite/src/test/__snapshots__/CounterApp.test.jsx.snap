// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CounterApp tarea debe hacer match con el snapshot 1`] = `undefined`;
